src/main/java/org/reservationapplication/repository/CoworkingSpaceRepository.java

```diff
@@ -0,0 +1,78 @@
+package org.reservationapplication.repository;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.reservationapplication.exeption.CoworkingSpaceNotFoundException;
+import org.reservationapplication.exeption.CoworkingStorageException;
+import org.reservationapplication.model.CoworkingSpace;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CoworkingSpaceRepository {
+    private static final String COWORKING_FILE_NAME = "coworking_spaces.json";
+    private ObjectMapper objectMapper;
+    private static long nextId = 0L;
+
+    public CoworkingSpaceRepository() {
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public static long getNextID(){
+        return nextId++;
+    }
+
+    public void saveCoworkingSpace(List<CoworkingSpace> generalCoworkingSpace) {
+        File file = new File(COWORKING_FILE_NAME);
+
+        try {
+            objectMapper.writeValue(file, generalCoworkingSpace);
+        } catch (IOException e) {
+            throw new CoworkingStorageException("Failed to save coworking spaces to file", e);
+        }
+    }
+
+    public List<CoworkingSpace> readCoworkingSpace() {
+        File file = new File(COWORKING_FILE_NAME);
+        List<CoworkingSpace> generalCoworkingSpace;
+
+        try {
+            if (file.exists() && file.length() > 0) {
+                generalCoworkingSpace = objectMapper.readValue(file, new TypeReference<>(){});
```

Comment:
reading it every time is not good practice. could you try to create cash? (saving and rereading not avery time, but some time after). It's not mandatory, but try, I see you love complicated stuff)